src/librbd/ImageCtx.cc

```diff
@@ -59,7 +59,7 @@ class ThreadPoolSingleton : public ThreadPool {
                  "rbd_op_threads") {
     start();
   }
-  virtual ~ThreadPoolSingleton() {
+  ~ThreadPoolSingleton() {
```

Comment:
Nit: since the "override" keyword isn't being applied, I would prefer to keep the "virtual" keyword on virtual destructors to highlight that fact.